hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/metrics/FSDatasetMBean.java

```diff
@@ -122,4 +122,9 @@ public interface FSDatasetMBean extends MetricsSource {
    * Returns the number of blocks that the datanode was unable to uncache
    */
   public long getNumBlocksFailedToUncache();
+
+  /**
+   * Returns the last time in milliseconds of directory scanner run.
+   */
+  public long getLastDirectoryScannerRunTime();
```

Comment:
This name will be little misleading if it is the begin or end time of one loop of DirectoryScanner.

Reply:
Sir. have changed to getLastDirectoryScannerFinishTime and other fields have also modified. Thanks for this review.